subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -133,6 +136,17 @@ static struct aws_iot_suback_confirmation {
 
 static K_SEM_DEFINE(connection_poll_sem, 0, 1);
 
+#if defined(CONFIG_AWS_IOT_CONNECTION_POLL_THREAD)
+#ifdef CONFIG_BOARD_QEMU_X86
+#define POLL_THREAD_STACK_SIZE 4096
+#else
+#define POLL_THREAD_STACK_SIZE 3072
+#endif /* CONFIG_BOARD_QEMU_X86 */
+static struct k_thread aws_connection_poll_thread;
+K_THREAD_STACK_DEFINE(poll_thread_stack, POLL_THREAD_STACK_SIZE);
+void aws_iot_cloud_poll(struct aws_iot_config *const config);
+#endif /* CONFIG_AWS_IOT_CONNECTION_POLL_THREAD */
+
```

Comment:
I don't think those changes were intended for this PR?

Reply:
This was moved to the top because I created the thread run-time in the `aws_iot_init` function. I did however change the overall approach now so this is no longer necessary.